circleci_test.go

```diff
@@ -101,12 +104,28 @@ func TestClient_request(t *testing.T) {
 		fmt.Fprint(w, `{"login": "jszwedko"}`)
 	})
 
-	err := client.request("GET", "/me", &User{}, nil, nil)
+	err := client.request(context.Background(), "GET", "/me", &User{}, nil, nil)
 	if err != nil {
 		t.Errorf(`Client.request("GET", "/me", &User{}, nil, nil) errored with %s`, err)
 	}
 }
 
+func TestClientWithContext_request(t *testing.T) {
+	setup()
+	defer teardown()
+	client.Token = "ABCD"
+	mux.HandleFunc("/me", func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(1 * time.Second)
+	})
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Microsecond)
+	defer cancel()
+	err := client.request(ctx, "GET", "/me", &User{}, nil, nil)
+	if err == nil || !errors.As(err, &context.DeadlineExceeded) {
```

Comment:
Sorry, I was a bit wrong. `context.DeadlineExceeded` is a variable. So `if err == nil || !errors.Is(err, context.DeadlineExceeded) {` is more suited here.

Sorry for the double-work.

Reply:
Done, no problem. Also, I'm learning hehehe